### PR TITLE
Upgrade to nix 0.26

### PR DESCRIPTION
**Pull request by fasterthanlime**
As discussed in https://github.com/bytecodealliance/userfaultfd-rs/issues/40 (almost missed that 👍 emoji reaction!)

### CODE REVIEW

File: examples/manpage.rs [hunk 1]
```diff
@@ -3,9 +3,8 @@ use libc::{self, c_void};
 use nix::poll::{poll, PollFd, PollFlags};
 use nix::sys::mman::{mmap, MapFlags, ProtFlags};
 use nix::unistd::{sysconf, SysconfVar};
-use std::env;
 use std::os::unix::io::AsRawFd;
-use std::ptr;
+use std::{convert::TryInto, env};
```
Comment (fasterthanlime) [inline examples/manpage.rs:7]: Note that if the crate was edition 2021 we wouldn't need to explicitly import `TryInto`: https://doc.rust-lang.org/edition-guide/rust-2021/prelude.html

**Comment by pchickey** [comment]
Thanks! I will leave a more obvious issue response next time than the `+1` emoji.

I think the CI issue is easy enough to fix, i'll do that on another PR right now...

File: examples/manpage.rs (hunk 1 above)
Comment (pchickey) [reply examples/manpage.rs:7]: Thanks. If you'd like to follow up with a PR to upgrade to edition 2021 that would be fine, but its not a big worry either way.

**Comment by pchickey** [comment]
Should be fixed now - please try merging in main

**Comment by fasterthanlime** [comment]
Thanks for fixing CI! It passes now.

**Review by pchickey — APPROVED** [review]

**Comment by pchickey** [comment]
Published https://crates.io/crates/userfaultfd/0.5.1

**Pull request merged**